src/backend/distributed/shardsplit/shardsplit_decoder.c

```diff
@@ -129,6 +134,40 @@ split_change_cb(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
 	}
 
 	Relation targetRelation = RelationIdGetRelation(targetRelationOid);
+
+	/*
+	 * If any columns from source relation have been dropped, then the tuple needs to
+	 * be formatted according to the target relation.
+	 */
+	TupleDesc sourceRelationDesc = RelationGetDescr(relation);
+	TupleDesc targetRelationDesc = RelationGetDescr(targetRelation);
+	if (sourceRelationDesc->natts > targetRelationDesc->natts)
+	{
+		HeapTuple targetRelationTuple = GetTupleForTargetSchema(existingTuple,
+																sourceRelationDesc,
+																targetRelationDesc);
+		switch (change->action)
+		{
+			case REORDER_BUFFER_CHANGE_INSERT:
+			{
+				change->data.tp.newtuple->tuple = *targetRelationTuple;
+				break;
+			}
+
+			case REORDER_BUFFER_CHANGE_UPDATE:
+			{
+				change->data.tp.newtuple->tuple = *targetRelationTuple;
+				break;
+			}
+
+			case REORDER_BUFFER_CHANGE_DELETE:
+			{
+				change->data.tp.oldtuple->tuple = *targetRelationTuple;
```

Comment:
does this still work correctly when not using replica identity full?

Reply:
For the testcase, I have not used full replica identity. Could you suggest a testcase?